File: src/main/scala/ru/tinkoff/gatling/amqp/request/PublishDslBuilderMessage.scala
package ru.tinkoff.gatling.amqp.request

import java.nio.charset.Charset

import io.gatling.core.config.GatlingConfiguration
import io.gatling.core.session.Expression
import ru.tinkoff.gatling.amqp.action.PublishBuilder

case class PublishDslBuilderMessage(
    requestName: Expression[String],
    destination: AmqpExchange,
    configuration: GatlingConfiguration
) {

  def textMessage(text: Expression[String], charset: Charset = configuration.core.charset): PublishDslBuilder =
    message(TextAmqpMessage(text, charset))
  def bytesMessage(bytes: Expression[Array[Byte]]): PublishDslBuilder = message(BytesAmqpMessage(bytes))

  private def message(mess: AmqpMessage) =
    PublishDslBuilder(
      AmqpAttributes(requestName, destination, None, mess),
      PublishBuilder(_, configuration)
    )

}

File: src/main/java/ru/tinkoff/gatling/javaapi/PublishDslBuilderMessage.java
package ru.tinkoff.gatling.javaapi;

import java.nio.charset.Charset;
import static io.gatling.javaapi.core.internal.Expressions.*;


public class PublishDslBuilderMessage {
    private final ru.tinkoff.gatling.amqp.request.PublishDslBuilderMessage wrapped;

    public PublishDslBuilderMessage(ru.tinkoff.gatling.amqp.request.PublishDslBuilderMessage wrapped){
        this.wrapped = wrapped;
    }
    public PublishDslBuilder textMessage(String text){
        return textMessage(text, io.gatling.core.Predef.configuration().core().charset());
    }
    public PublishDslBuilder textMessage(String text, Charset charset){
        return new PublishDslBuilder(wrapped.textMessage(toStringExpression(text), charset));
    }
}
